README.md
[![](https://img.shields.io/github/actions/workflow/status/alefragnani/vscode-jenkins-status/main.yml?branch=master)](https://img.shields.io/github/actions/workflow/status/alefragnani/vscode-jenkins-status/main.yml?branch=master)

<p align="center">
  <br />
  <a title="Learn more about Jenkins Status" href="https://github.com/eramitmittal/vscode-jenkins-status"><img src="https://raw.githubusercontent.com/alefragnani/vscode-jenkins-status/master/images/vscode-jenkins-status-logo-readme.png" alt="Read-only Logo" width="70%" /></a>
</p>

# What's new in Jenkins Status 4.5

* Adds **Localization** support
* Adds **Workspace Trust** support
* Adds **Virtual Workspace** support
* Adds **Remote Development** support
* Adds **Multi-root** support

## Support

**Jenkins Status** is an open source extension created for **Visual Studio Code**. While being free and open source, if you find it useful, please consider supporting it.

<table align="center" width="60%" border="0">
  <tr>
    <td>
      <a title="Paypal" href="https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=EP57F3B6FXKTU&lc=US&item_name=Alessandro%20Fragnani&item_number=vscode%20extensions&currency_code=USD&bn=PP%2dDonationsBF%3abtn_donate_SM%2egif%3aNonHosted"><img src="https://www.paypalobjects.com/en_US/i/btn/btn_donate_SM.gif"/></a>
    </td>
    <td>
      <a title="GitHub Sponsors" href="https://github.com/sponsors/alefragnani"><img src="https://raw.githubusercontent.com/alefragnani/oss-resources/master/images/button-become-a-sponsor-rounded-small.png"/></a>
    </td>
    <td>
      <a title="Patreon" href="https://www.patreon.com/alefragnani"><img src="https://raw.githubusercontent.com/alefragnani/oss-resources/master/images/button-become-a-patron-rounded-small.png"/></a>
    </td>
  </tr>
</table>

# Jenkins Status

It adds an area in the status bar, indicating the build status for any **Jenkins** job. Specially useful if you want to _connect_ a project with its CI server. 

# Features

View the Jenkins build status of your project inside Visual Studio Code.

![screenshot](images/jenkins-screenshot.png)

It is automatically enabled if you have a `.jenkins` or `.jenkinsrc.js` file in the root folder of your project. The only required information is the `url` pointing to your Jenkins job. 

If you need _authentication_, just add `username` and `password_or_token` in the `.jenkins` file and you are ready to go.

**.jenkins** file

```json
{
    "url": "http://127.0.0.1:8080/job/myproject/",
    "username": "jenkins_user",
    "password": "jenkins_password_or_token"
}
``` 
or
```json
[
    {
        "url": "http://127.0.0.1:8080/job/myproject/",
        "name": "Jenkins Build",
        "username": "jenkins_user",
        "password": "jenkins_password_or_token"
    },
    {
        "url": "http://127.0.0.1:8080/job/myprojectTests/",
        "name": "Jenkins Acceptance Tests",
        "username": "jenkins_user",
        "password": "jenkins_password_or_token"
    }
]
``` 
**.jenkinsrc.js** file
```js
// can also return a promise of required JSON structure
module.exports = [{
    "url": "http://127.0.0.1:8080/job/myproject/",
    "name": "Jenkins Build",
    "username": "jenkins_user",
    "password": "jenkins_password_or_token"
},
{
    "url": "http://127.0.0.1:8080/job/myprojectTests/",
    "name": "Jenkins Acceptance Tests",
    "username": "jenkins_user",
    "password": "jenkins_password_or_token"
}];
```

If you are having trouble with self-signed certificates and your build status says `SELF_SIGNED_CERT_IN_CHAIN`, you could use a _workaroud_ adding a `strictTls` flag to your `.jenkins` file or `.jenkinsrc.js` export:

```json
    "strictTls": false
```

## Available commands

* `Jenkins: Open in Jenkins:` Open the Jenkins project in you browser 
* `Jenkins: Open in Jenkins (Console Output):` Open the Console Output of the Jenkins project in you browser 
* `Jenkins: Update Status:` Manually update the status of our Jenkins project

## Working with Remotes

The extension support [Remote Development](https://code.visualstudio.com/docs/remote/remote-overview) scenarios, and you may choose how to use it, depending on your needs

### I access Remotes, and use `.jenkins` files

This is the _regular_ scenario, and that's why you don't need to do anything special for the extension to work. It works out of the box.

When installed locally, the extension will properly recognize the `.jenkins` file on remotes, and display the status bar for each Url.

_It just works_

### I access Remotes, but I use `.jenkinsrc.js` files

If you need to use `.jenkinsrc.js` files, the extension must be installed on that remote. This happens because it is not possible to _import_ the `.jenkinsrc.js` file remotely.

## Available settings

* Interval (in minutes) to automatically update the status
```json
    "jenkins.polling": 2
```
> Note: 0 (zero) means _no update_

## Contributors

Special thanks to the people that have contributed to the project:

* Kalin Krustev (@md2perpe) - Cleanup source ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/97))
* Kalin Krustev (@md2perpe) - Typo ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/89))
* Kalin Krustev (@kalinkrustev) - Support Remote Development ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/76))
* Keunhong Lee (@leeopop) - Skip authentication when no `username` is provided ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/35))
* Amit Mittal (@eramitmittal) - Multiple Jobs and in-progress status support ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/17))
* Kimitaka Watanabe (@kimitaka) - Updated `.jenkins` example in README  ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/31))
* Zhao Xiaojie (@LinuxSuRen) - Status Bar improvements ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/29))
* Grzegorz Judas (@grzegorzjudas) - HTTPS Support using `NODE_TLS_REJECT_UNAUTHORIZED` environment variable ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/23))
* Adam Parkin (@pzelnip) - Fix tooltip for failed builds ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/15))
* Mike Patrick (@mikepatrick) and Mensa Ulysse (@umens) - Support Authentication ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/10))
* Vojtěch Habarta (@vojtechhabarta) - Support for larger JSON responses ([see PR](https://github.com/alefragnani/vscode-jenkins-status/pull/7))

Also thanks to everyone who helped opening issues with ideas and bug reports.

# License

[MIT](LICENSE.md) &copy; Alessandro Fragnani

CHANGELOG.md
## [4.5.0] - 2023-07-27
### Added
- Localization (l10n) support (issue [#90](https://github.com/alefragnani/vscode-jenkins-status/issues/90))

### Changed
- Avoid What's New when using Gitpod (issue [#100](https://github.com/alefragnani/vscode-jenkins-status/issues/100))
- Avoid What's New when installing lower versions (issue [#100](https://github.com/alefragnani/vscode-jenkins-status/issues/100))

### Fixed
- Typo (PR [#89](https://github.com/alefragnani/vscode-jenkins-status/pull/89) - kudos to @md2perpe)

### Internal
- Update badges in README (issue [#102](https://github.com/alefragnani/vscode-jenkins-status/issues/102))
- Security Alert: word-wrap (dependabot [PR #101](https://github.com/alefragnani/vscode-jenkins-status/pull/101))
- Security Alert: webpack (dependabot [PR #99](https://github.com/alefragnani/vscode-jenkins-status/pull/99))
- Cleanup source (PR [#97](https://github.com/alefragnani/vscode-jenkins-status/pull/97) - kudos to @md2perpe)
- Support Implicit Activation Events API (issue [#94](https://github.com/alefragnani/vscode-jenkins-status/issues/94))
- Security Alert: minimatch (dependabot [PR #92](https://github.com/alefragnani/vscode-jenkins-status/pull/92))
- Security Alert: qs (dependabot [PR #91](https://github.com/alefragnani/vscode-jenkins-status/pull/91))
- Security Alert: terser (dependabot [PR #85](https://github.com/alefragnani/vscode-jenkins-status/pull/85))
- Package Cleanup (issue [#84](https://github.com/alefragnani/vscode-jenkins-status/issues/84))

## [4.4.1] - 2022-07-17
### Internal
* Add GitHub Sponsors support (PR [#83](https://github.com/alefragnani/vscode-jenkins-status/pull/83))

## [4.4.0] - 2022-03-30
### Added
* Support **Remote Development** (issue [#41](https://github.com/alefragnani/vscode-jenkins-status/issues/41))
* Support **Workspace Trust** (issue [#66](https://github.com/alefragnani/vscode-jenkins-status/issues/66))
* Support **Virtual Workspaces** (issue [#67](https://github.com/alefragnani/vscode-jenkins-status/issues/67))
* Support new Status Bar API (issue [#68](https://github.com/alefragnani/vscode-jenkins-status/issues/68))

## [4.3.0] - 2021-11-13
### Internal
- Add CONTRIBUTING documentation (issue [#71](https://github.com/alefragnani/vscode-jenkins-status/issues/71))
- Update dependencies (issue [#73](https://github.com/alefragnani/vscode-jenkins-status/issues/73))
- Security Alert: lodash (dependabot [PR #65](https://github.com/alefragnani/vscode-jenkins-status/pull/65))
- Security Alert: ssri (dependabot [PR #64](https://github.com/alefragnani/vscode-jenkins-status/pull/64))
- Security Alert: y18n (dependabot [PR #63](https://github.com/alefragnani/vscode-jenkins-status/pull/63))
- Security Alert: elliptic (dependabot [PR #62](https://github.com/alefragnani/vscode-jenkins-status/pull/62))

## [4.2.1] - 2020-09-05
### Fixed
- `Open in Jenkins (Console Output)` command with unnamed job (issue [#56](https://github.com/alefragnani/vscode-jenkins-status/issues/56))

## [4.2.0] - 2020-08-08
### Internal
- Use `vscode-ext-codicons` package (issue [#52](https://github.com/alefragnani/vscode-jenkins-status/issues/52))
- Shrink installation size/time (issue [#51](https://github.com/alefragnani/vscode-jenkins-status/issues/51))

### Fixed
- Security Alert: elliptic (dependabot [PR #54](https://github.com/alefragnani/vscode-jenkins-status/pull/54))
- Security Alert: lodash (dependabot [PR #53](https://github.com/alefragnani/vscode-jenkins-status/pull/53))

## [4.1.0] - 2020-06-14
## Added
- Localization support (issue [#30](https://github.com/alefragnani/vscode-jenkins-status/issues/30))

### Internal
- Support Extension View Context Menu (issue [#50](https://github.com/alefragnani/vscode-jenkins-status/issues/50))
- Migrate from TSLint to ESLint (issue [#47](https://github.com/alefragnani/vscode-jenkins-status/issues/47))

## [4.0.0] - 2020-02-22
### Added
- **Multiple Jobs** and **in-progress** status support (Thanks to @eramitmittal [PR #17](https://github.com/alefragnani/vscode-jenkins-status/pull/17))
- Auto-detect changes in `.jenkins` file

### Changed
- **Status Bar** tooltip

### Fixed
- Skip authentication when no `username` is provided (Thanks to @leeopop [PR #35](https://github.com/alefragnani/vscode-jenkins-status/pull/35))

## [3.1.2] - 2019-05-28
### Fixed
- Security Alert: tar

## [3.1.1] - 2019-03-14
### Fixed
- What's New page broken in VS Code 1.32 due to CSS API changes
- Updated `.jenkins` example in README (Thanks to @kimitaka [PR #31](https://github.com/alefragnani/vscode-jenkins-status/pull/31))

## [3.1.0] - 2019-01-18
### Added
- Status Bar improvements (Thanks to @LinuxSuRen [PR #29](https://github.com/alefragnani/vscode-jenkins-status/pull/29))

## [3.0.0] - 2018-12-02
### Added
- What's New

## [2.2.0] - 2018-04-16
### Added
- Patreon button

## [0.6.1 - 2.1.1] - 2018-04-13
### Fixed
- Incorrect value in `strictTls` sample

## [0.6.0 - 2.1.0] - 2018-04-13
### Added
- HTTPS Support using `NODE_TLS_REJECT_UNAUTHORIZED` environment variable (PR [#23](https://github.com/alefragnani/vscode-jenkins-status/pull/23) - kudos to @grzegorzjudas)

## [0.5.1 - 2.0.1] - 2017-11-13
### Fixed
- Error while opening single-root workspace with no active file

## [0.5.0 - 2.0.0] - 2017-10-28
### Added
- Multi-root support (issue [#18](https://github.com/alefragnani/vscode-jenkins-status/issues/18))

## [0.4.1 - 1.3.1] - 2017-10-27
### Fixed
- Fix tooltip for failed builds (PR [#15](https://github.com/alefragnani/vscode-jenkins-status/pull/15) - kudos to @pzelnip)

## [0.4.0 - 1.3.0] - 2017-06-10
### Added
- Support Authentication (PR [#10](https://github.com/alefragnani/vscode-jenkins-status/pull/10) - kudos to @mikepatrick and @umens)

## [0.3.0 - 1.2.0] - 2017-01-03
### Added
- New Command `Open in Jenkins` (Console Output)

## [0.2.0 - 1.1.0] - 2016-12-29
### Added
- Polling for automatic status update (issue [#1](https://github.com/alefragnani/vscode-jenkins-status/issues/1))

## [0.1.2 - 1.0.1] -2016-12-26 
### Fixed

src/whats-new/contentProvider.ts
/*---------------------------------------------------------------------------------------------
*  Copyright (c) Alessandro Fragnani. All rights reserved.
*  Licensed under the MIT License. See License.md in the project root for license information.
*--------------------------------------------------------------------------------------------*/

// tslint:disable-next-line:max-line-length
import { ChangeLogItem, ChangeLogKind, ContentProvider, Header, Image, IssueKind, SocialMediaProvider, SupportChannel } from "../../vscode-whats-new/src/ContentProvider";

export class JenkinsStatusContentProvider implements ContentProvider {
    
    public provideHeader(logoUrl: string): Header {
        return <Header> {logo: <Image> {src: logoUrl, height: 50, width: 50}, 
            message: `<b>Jenkins Status</b> adds an area in the status bar, indicating the 
            <b>build status</b> for any <b>Jenkins</b> job. Specially useful if you want to 
            <i>connect</i> a project with its CI server`};
    }

    public provideChangeLog(): ChangeLogItem[] {
        const changeLog: ChangeLogItem[] = [];

        changeLog.push({ kind: ChangeLogKind.VERSION, detail: { releaseNumber: "4.5.0", releaseDate: "June 2023" } });
        changeLog.push({
            kind: ChangeLogKind.NEW,
            detail: {
                message: "Add <b>Localization (l10n)</b> support",
                id: 90,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.CHANGED,
            detail: {
                message: "Avoid What's New when using Gitpod",
                id: 100,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.CHANGED,
            detail: {
                message: "Avoid What's New when installing lower versions",
                id: 100,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.FIXED,
            detail: {
                message: "Typo",
                id: 89,
                kind: IssueKind.PR,
                kudos: "@md2perpe"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "README",
                id: 102,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Security Alert: word-wrap",
                id: 101,
                kind: IssueKind.PR,
                kudos: "dependabot"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Security Alert: webpack",
                id: 99,
                kind: IssueKind.PR,
                kudos: "dependabot"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Cleanup source",
                id: 97,
                kind: IssueKind.PR,
                kudos: "@md2perpe"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Support Implicit Activation Event API",
                id: 94,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Security Alert: minimatch",
                id: 92,
                kind: IssueKind.PR,
                kudos: "dependabot"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Security Alert: qs",
                id: 91,
                kind: IssueKind.PR,
                kudos: "dependabot"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Security Alert: terser",
                id: 85,
                kind: IssueKind.PR,
                kudos: "dependabot"
            }
        });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: {
                message: "Package cleanup",
                id: 84,
                kind: IssueKind.Issue
            }
        });

        changeLog.push({ kind: ChangeLogKind.VERSION, detail: { releaseNumber: "4.4.1", releaseDate: "June 2022" } });
        changeLog.push({
            kind: ChangeLogKind.INTERNAL,
            detail: "Add <b>GitHub Sponsors</b> support"
        });

        changeLog.push({ kind: ChangeLogKind.VERSION, detail: { releaseNumber: "4.4.0", releaseDate: "March 2022" } });
        changeLog.push({
            kind: ChangeLogKind.NEW,
            detail: {
                message: "Support <b>Remote Development</b>",
                id: 41,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.NEW,
            detail: {
                message: "Support <b>Workspace Trust</b>",
                id: 66,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.NEW,
            detail: {
                message: "Support <b>Virtual Workspace</b>",
                id: 67,
                kind: IssueKind.Issue
            }
        });
        changeLog.push({
            kind: ChangeLogKind.NEW,
            detail: {
                message: "Support new Status Bar API",
                id: 68,
                kind: IssueKind.Issue
            }
        });

        return changeLog;
    }

    public provideSupportChannels(): SupportChannel[] {
        const supportChannels: SupportChannel[] = [];
        supportChannels.push({
            title: "Become a sponsor on GitHub",
            link: "https://www.github.com/sponsors/alefragnani",
            message: "Become a Sponsor"
        });
        supportChannels.push({
            title: "Donate via PayPal",
            link: "https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=EP57F3B6FXKTU&lc=US&item_name=Alessandro%20Fragnani&item_number=vscode%20extensions&currency_code=USD&bn=PP%2dDonationsBF%3abtn_donate_SM%2egif%3aNonHosted",
            message: "Donate via PayPal"
        });
        return supportChannels;
    }
}

export class JenkinsStatusSocialMediaProvider implements SocialMediaProvider {
    public provideSocialMedias() {
        return [{
            title: "Follow me on Twitter",
            link: "https://www.twitter.com/alefragnani"
        }];
    }
}